### PR TITLE
Update supported python versions

### PR DESCRIPTION
**Pull request by altheaden**
Python version 3.8 has now been dropped from conda-forge, and 3.12 has been added. This PR just updates the related checks and files to be consistent with those supported versions.

Checklist
* [x] Developer's Guide has been updatedbuilding_docs.html) and changes look as expected
* [ ] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

**Comment by altheaden** [comment]
@xylar as you will see in the changed files, I deleted and added some files in the `ci` folder. I just guessed based on the existing files that the only change within these files would be the version number, but you should definitely look that over in case it should be different. Also, please take a look at my change in the developer's guide, it's very small but I want to make sure that it's accurate.

**Comment by altheaden** [comment]
## Testing
I haven't done any testing yet, as I'm not sure if any of these changes would affect building/running compass. I am happy to run the PR suite if that's desired, but I'll hold off for now.

File: docs/developers_guide/quick_start.rst [hunk 1]
```diff
@@ -217,7 +217,7 @@ Optional flags
     Check to make sure expected commands are present
 
 ``--python``
-    Select a particular python version (the default is currently 3.8)
+    Select a particular python version (the default is currently 3.9)
```
Comment (xylar) [inline docs/developers_guide/quick_start.rst:220]:
```suggestion
    Select a particular python version (the default is currently 3.12)
```

**Review by xylar — COMMENTED** [review]
Wonderful!  One tiny change.

File: docs/developers_guide/quick_start.rst (hunk 1 above)
Comment (xylar) [reply docs/developers_guide/quick_start.rst:220]: This was *very* out of date.

**Comment by xylar** [comment]
@altheaden, these could in theory affect tests but I'm not worried.  As long as CI passes, I think we can consider that sufficient testing.

**Comment by altheaden** [comment]
@xylar Done!

**Comment by xylar** [comment]
Thanks @altheaden!

**Pull request merged**